client/my-sites/stats/stats-module-utm-builder/stats-modeule-utm-builder-form.tsx

```diff
@@ -91,7 +91,9 @@ const UtmBuilder: React.FC = () => {
 
 	const utmString = ! url
 		? translate( 'Fill out campaign parameters to see the URL' )
-		: `${ url }${ campaignString ? `/?${ campaignString }` : '' }`;
+		: `${ url }${
```

Comment:
I think the URL concatenating works well here; however it would end up with URLs not looking as very nice. For example, if it's an URL ends with a question mark, the output of the expression would be like `https://abc.com/?&utm_source=xxx...`. It is still legal, but it doesn't look good.

I wonder if we could parse the URL with `new URL(xxx)`, alter the `searchParams` and then serialize it again, which would generate a fully working URL without any lingering characters.

Reply:
This is a nice idea, but it might be more complex than we want. If users type the URL, we might quickly encounter an invalid URL. We would have to debounce the keystrokes, and this is not perfect either.
Did you have something else in mind?

Reply:
We ship it as is now. I think it'll be great to surface any illegal URLs rather than generating one for them. Again, I think it would work well for now, and we could leave it for future improvements.